python/llm/src/bigdl/llm/transformers/layers/rope_embedding.py

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import torch
+import logging
+from bigdl.llm.transformers.xpu_customize_fwd import custom_fwd, custom_bwd
+from bigdl.llm.utils.common import invalidInputError
+
+LOG = logging.getLogger("bigdl.llm.rope_embedding")
+
+
+# Fast RoPE for finetuning, split the q and k
+def apply_fast_rope_embedding(q, k, position_ids, model_family):
+    if q.device.type != "xpu":
+        invalidInputError(False,
+                          f"only xpu is supported in this function")
+    if model_family in ["llama", "baichuan", "internlm", "aquila", "gpt_neox", "mistral",
+                        "mixtral"]:
+        q_embed = Fast_RoPE_Embedding.apply(q, position_ids)
+        k_embed = Fast_RoPE_Embedding.apply(k, position_ids)
+        return q_embed, k_embed
+    else:
+        invalidInputError(False,
+                          f"{model_family} is not supported.")
+
+
+# Fast RoPE for finetuning, split the q and k
+class Fast_RoPE_Embedding(torch.autograd.Function):
```

Comment:
Nitpick: How about just `FastRopeEmbedding`?

Reply:
Yes, it seems a big strange, updated.